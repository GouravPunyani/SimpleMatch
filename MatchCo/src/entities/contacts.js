// @flow
import { observable } from 'mobx'

export class Contacts {
  @observable customerServiceEmail: ?string
  @observable feedbackEmail: ?string
  @observable mediaEmail: ?string
  @observable customerServicePhone: ?string
  @observable facebookUrl: ?string
  @observable instagramUrl: ?string
  @observable twitterUrl: ?string
  @observable companyUrl: ?string
}

export const createDefault = () => {
  const contacts = new Contacts()
  contacts.customerServiceEmail = 'made2fit@bareminerals.com'
  contacts.feedbackEmail = 'made2fit@bareminerals.com'
  contacts.mediaEmail = 'made2fit@bareminerals.com'
  contacts.customerServicePhone = '800.605.8816'
  contacts.facebookUrl = 'https://www.facebook.com/bareminerals?fref=ts'
  contacts.instagramUrl = 'https://instagram.com/bareminerals'
  contacts.twitterUrl = 'https://twitter.com/bareminerals'
  contacts.companyUrl = 'http://www.bareminerals.com'
  return contacts
}
