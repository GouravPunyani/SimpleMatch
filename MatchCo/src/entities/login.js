// @flow
import { observable, action, computed } from 'mobx'
import Validate from 'validate.js'
import I18n from 'react-native-i18n'
import { memoize, forEach, isNil, curry, keys, isEmpty } from 'ramda'

// gets the constraints, but delayed in a function due to a timing issue
// with i18n.
const getConstraints = memoize(() => {
  return {
    email: {
      presence: { message: I18n.t('errors.invalidEmail') },
      email: { message: I18n.t('errors.invalidEmail') }
    },
    password: {
      presence: { message: I18n.t('errors.passwordRequired') }
    }
  }
})

export class Login {
  @observable email: string
  @observable password: string

  @observable errors: any = {
    email: null,
    password: null
  }

  /**
   * Updates a field key with this new value.
   */
  update = action(curry(
    (key: string, value: string) => {
      this[key] = value
      this.validate(key)
    }
  ))

  @action
  async validate (...fields: string[]) {
    const currentErrors = this.errors
    const isSubset = fields.length > 0

    // grab all the errors
    const errors = Validate(this, getConstraints(), { fullMessages: false }) || {}

    // determine which fields to update
    const fieldsToUpdate = isEmpty(fields) || keys(currentErrors)

    // for subsets, reset any errors that may be error-free now
    if (isSubset && !isEmpty(errors)) {
      forEach(field => {
        if (isNil(errors[field])) currentErrors[field] = null
      }, fields)
    }

    // set each error
    forEach(field => {
      currentErrors[field] = errors[field]
    }, fieldsToUpdate)

    return isEmpty(errors)
  }

  @computed get isValid () {
    return isNil(Validate(this, getConstraints()))
  }
}
