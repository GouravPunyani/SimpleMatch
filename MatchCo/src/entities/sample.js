/**
 * A sample is a single color calibration line.
 */
export type Sample = {
  'sample_Red': number,
  'sample_Green': number,
  'sample_Blue': number,
  'sample_L': number,
  'sample_A': number,
  'sample_B': number,
  sampleLum?: number,
  specLum?: number
}
