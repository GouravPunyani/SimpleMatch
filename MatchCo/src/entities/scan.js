import type { Sample } from './sample'

/**
 * A scan is when the user performs an end-to-end scan of their face.
 */
export type Scan = {
  /** The full path to the saved scan file. */
  filename: string,
  /** The date of the scan in UTC. */
  date: Date,
  /** The device model used */
  device: string,
  /** The protocol version of the scan file. */
  version: string,
  /** The parsed sample data. */
  samples: Array<Sample>,
  /** The unparsed contents of the scan file. */
  raw: string
}
