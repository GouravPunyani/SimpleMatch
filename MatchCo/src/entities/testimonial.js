import { observable } from 'mobx'

export class Testimonial {
  @observable id: string
  @observable name: string
  @observable image: ?string
  @observable quote: string
  @observable videoUrl: ?string
}
