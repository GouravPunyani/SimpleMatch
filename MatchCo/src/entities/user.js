import { observable } from 'mobx'

/**
 * The known list of reasons for being unable to login.
 */
export type LoginErrorCode = 'wrong' | 'unknown';

export class User {
  @observable uid: ?string
  @observable email: ?string
  @observable displayName: ?string = 'Default Username' // TODO(steve): escalate
  @observable photoURL: ?string
  @observable fetching: boolean = false
  @observable errorCode: LoginErrorCode
  @observable scanTint: string = 'hotpink'
  @observable scanDate: string = 'May 32, 2017'
}
