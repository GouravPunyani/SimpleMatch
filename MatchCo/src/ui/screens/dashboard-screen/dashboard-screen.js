// @flow

import type { UserStore } from '../../../stores/user-store'
import type { NavigationStore } from '../../../stores/navigation-store'
import { Dimensions } from 'react-native'
import { inject, observer } from 'mobx-react'
import React, { Component } from 'react'
import { StaticLayout, Button, ButtonPanel, Text, DashboardCarousel } from '../..'
import { translator } from '../../../services/translate/translator'
import { palette, spacing } from '../../theme'
const { width, height } = Dimensions.get('window')

export type Props = {
  userStore: UserStore,
  navigationStore: NavigationStore,
  translate: (key: string) => string
};

@translator
@inject('userStore', 'navigationStore')
@observer
export class DashboardScreen extends Component<*, Props, *> {
  textForLoggingIn () {
    const { translate, userStore } = this.props
    const { isLoggedIn, user } = userStore
    const { email, fetching } = user

    if (fetching) return translate('dashboard.loggingIn')
    return isLoggedIn ? email : translate('dashboard.notLoggedIn')
  }

  render () {
    const { navigationStore, translate } = this.props
    return (
      <StaticLayout style={styles.wrapper}>
        <DashboardCarousel/>
        <ButtonPanel>
          <Button
            text={translate('dashboard.getStarted')}
            onPress={() => navigationStore.navigateTo('product')}
            style={styles.button}
          />
        </ButtonPanel>
      </StaticLayout>
    )
  }
}

const styles = {
  wrapper: { flex:1, padding: spacing.none },
  button: {
    position: 'absolute',
    left: width*.05,
    right: width*.05,
    bottom: height*.15
  }
}
