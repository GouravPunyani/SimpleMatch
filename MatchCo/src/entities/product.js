import { observable } from 'mobx'
import { Testimonial } from './testimonial'

export class Product {
  @observable id: string
  @observable name: string
  @observable ingredients: string
  @observable description: string
  @observable price: number
  @observable sizeImperial: string
  @observable sizeMetric: string
  @observable testimonials: Array<Testimonial> = []
}
