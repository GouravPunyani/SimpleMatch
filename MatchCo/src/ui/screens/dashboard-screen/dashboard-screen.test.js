// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { User } from '../../../entities/user'
import { DashboardScreen as Screen } from './dashboard-screen'

test('renders correctly', () => {
  const user = new User()
  user.email = 'steve@infinite.red'
  user.fetching = false
  user.uid = '123'
  const userStore = { user }

  const navigationState = {}
  const navigationStore = { navigationState }

  const tree = renderer.create(
    <Screen userStore={userStore} navigationStore={navigationStore} />
  )
  expect(tree).toMatchSnapshot()
})
