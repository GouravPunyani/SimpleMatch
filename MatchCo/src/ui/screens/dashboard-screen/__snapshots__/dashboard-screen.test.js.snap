// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
      "justifyContent": "flex-start",
      "padding": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Nexa",
          "fontSize": 14,
        },
      ]
    }
  >
    Not Logged In
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "margin": 4,
          "minHeight": 44,
          "minWidth": 44,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
          ]
        }
      >
        Proceed To Checkout
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "white",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "margin": 4,
          "minHeight": 44,
          "minWidth": 44,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
          ]
        }
      >
        Referrals
      </Text>
    </View>
  </View>
</View>
`;
